### PR TITLE
Move ReluQuantFusion to Level2 for CPU EP only

### PR DESCRIPTION
**Pull request by adrianlizarraga**
### Description
Moves the `Relu -> QuantizeLinear` fusion to Level2 optimizations for CPU EP only.

### Motivation and Context
See the related PR for motivation and context: https://github.com/microsoft/onnxruntime/pull/20627

### CODE REVIEW

File: onnxruntime/test/optimizer/qdq_transformer_test.cc [hunk 1]
```diff
@@ -2763,6 +2763,57 @@ TEST(QDQTransformerTests, Clip) {
...
+      builder.AddQuantizeLinearNode<int8_t>(relu_output, 1.0f, zp, q_output);
+      builder.AddDequantizeLinearNode<int8_t>(q_output, 1.0f, zp, output_arg);
+    };
+
+    auto check_relu_graph = [&](InferenceSessionWrapper& session) {
+      auto op_to_count = CountOpsInGraph(session.GetGraph());
+      const QDQOpKeys qdq_keys = GetQDQOpKeys(false);
+      // Only fuse relu into Q if level >= 2 and zero_point == -128 for int8.
+      // Level1 graph:   input -> DQ -> Relu -> Q -> DQ -> output
+      // Level +2 graph: input -> DQ -> output (QuantReluFusion + QDQFinalCleanupTransformer transformers)
```
Comment (yihonglyu) [inline onnxruntime/test/optimizer/qdq_transformer_test.cc:2793]: Level2+?

Comment (adrianlizarraga) [reply onnxruntime/test/optimizer/qdq_transformer_test.cc:2793]: Yes, setting an optimization level of 3 (for example) also runs level 2 optimizations.

Comment (adrianlizarraga) [inline onnxruntime/test/optimizer/qdq_transformer_test.cc:2793]:
```suggestion
      // Level2+ graph: input -> DQ -> output (QuantReluFusion + QDQFinalCleanupTransformer transformers)
```

Comment (adrianlizarraga) [reply onnxruntime/test/optimizer/qdq_transformer_test.cc]: Oh I see that you were referring to the typo (`level +2` to `level 2+`). Fixed.

**Review by yihonglyu — APPROVED** [review]

**Pull request merged**